CIP-????/README.md

```diff
@@ -0,0 +1,530 @@
+preserved, despite the transaction body being broken into pieces. The same incentives apply for the
+transaction fee since the aggregator will want to minimize their own costs, and therefore, won't
+waste execution resources.
+
+### Transaction Piece Monoid Instance
+
+The monoid instance for transaction pieces defines the rules for combining the pieces. In order to
+be able to deterministically derive the full transaction, this instance must be standardized. The
+monoid instance requires answering two questions:
+
+1. What is an empty transaction piece?
+2. How are two transaction pieces combined?
+
+The first question is simple to answer: an empty transaction piece is an empty transaction body.
+Users fill out as much information as they wish when creating their transaction piece. The rest of
+the fields can remain empty.
+
+> [!IMPORTANT]
+> It doesn't really make sense to create a transaction piece with an input but no output, or an
+> output but no input. Therefore, these fields can remain required fields. The fee is also required,
+> but that can be set to zero in the pieces not paying the fee. All other fields in the transaction
+> body are already optional.
+
+The second question requires more consideration, but as a general rule, as long as there are no
+incompatibilities between transaction pieces, they can be combined into the full transaction. Any
+transaction that contains incompatible pieces should be rejected.
+
+#### Ordering
+
+The creator of the transaction piece gets to specify the order of inputs, outputs, reference inputs,
+and collateral inputs **within** their piece. Then, the aggregator (the one that batches pieces and
+submits the transaction) gets to decide the order **between** the pieces.
+
+For example, if Alice creates a piece with `input1` followed by `input2` and Bob creates a piece
+with `input3` followed by `input4`, then Charlie (the aggregator) only has two options:
+
+```
+[input1, input2, input3, input4] OR [input3, input4, input1, input2]
+```
+
+Charlie cannot put `input2` before `input1` because this would violate Alice's intent. The order of
+everything else is standardized (eg, token mints are sorted by name, withdrawals are sorted by
+credential, etc), so no one can control the order of them.
+
+> [!IMPORTANT] 
+> This would required [CIP-128](https://github.com/cardano-foundation/CIPs/pull/758) since the order
+> of the inputs and reference inputs would need to be preserved. There may be intents where the
+> order of these fields matter which is why the intent creator should get to specify their ordering.
+> The collateral inputs could be an exception, but if we are already making the change for inputs
+> and reference inputs, why not also make the change for collateral inputs?
+
+#### Duplicate Inputs
+
+It is valid for two separate transaction pieces in a batch to spend the same UTxOs (eg, a liquidity
+source UTxO). This UTxO can still only be spent once and the batch must still be fully balanced
```

Comment:
This not valid. Their position in the input set can affect the execution outcome of a plutus script that locks them